lib/iris/fileformats/pp.py

```diff
@@ -1642,12 +1641,8 @@ def __eq__(self, other):
             for attr in self.__slots__:
                 attrs = [hasattr(self, attr), hasattr(other, attr)]
                 if all(attrs):
-                    self_attr = getattr(self, attr)
-                    other_attr = getattr(other, attr)
-                    if isinstance(self_attr, biggus.NumpyArrayAdapter):
-                        self_attr = self_attr.concrete
-                    if isinstance(other_attr, biggus.NumpyArrayAdapter):
-                        other_attr = other_attr.concrete
+                    self_attr = as_concrete_data(getattr(self, attr))
+                    other_attr = as_concrete_data(getattr(other, attr))
```

Comment:
Well, I think we really want to be explicit here that we are realising the content.

In fact ```(self_attr == other+attr)``` ***is*** itself a lazy object, i.e. it hasn't looked at the data yet.
I fully expected that you would be *required* to call 'compute' + nothing else will do it (as with biggus),
however ***in fact*** it seems that applying np.all() will realise it + do the compare.
In my view that is not nice, and may even be a bug, as it's not documented anywhere -- see [comment on #2308](https://github.com/SciTools/iris/issues/2308#issuecomment-273499060)